lxd/storage_lvm.go

```diff
@@ -173,6 +173,47 @@ type storageLvm struct {
 	storageShared
 }
 
+func (s *storageLvm) getLvmBlockMountOptions() string {
+	if s.volume.Config["block.mount_options"] != "" {
+		return s.volume.Config["block.mount_options"]
+	}
+
+	if s.pool.Config["volume.block.mount_options"] != "" {
+		return s.pool.Config["volume.block.mount_options"]
+	}
+
+	if s.getLvmFilesystem() == "ext4" {
+		return "discard"
+	}
```

Comment:
That should be done with both ext4 and xfs, they both support the option and I believe in the past we would just always pass it.